contrib/mesos/pkg/executor/executor.go

```diff
@@ -208,23 +207,81 @@ func (k *KubernetesExecutor) isDone() bool {
 	}
 }
 
+func (k *KubernetesExecutor) NodeInfo() (ni NodeInfo) {
+	k.lock.RLock()
+	defer k.lock.RUnlock()
+
+	ni = k.nodeInfo
+	return
+}
+
+func (k *KubernetesExecutor) onRegister(ei *mesos.ExecutorInfo, si *mesos.SlaveInfo) {
+	k.lock.Lock()
+	defer k.lock.Unlock()
+
+	if !(&k.state).transition(disconnectedState, connectedState) {
+		log.Errorf("failed transitioning to a connected state")
+	}
```

Comment:
i thought "why don't we return here if the state doesn't match?". turns out that the underlying executor driver doesn't ever invoke Disconnected for any reason -- it makes no effort to communicate disconnection events back to the Executor. in apache/mesos, there's a libprocess `exited` func that's invoked when persistent socket connections are severed, mesos-go does no such thing. probably worth noting here, to some degree.